design-documents/storefront/configuraiton-propagation.md

```diff
@@ -0,0 +1,65 @@
+# Configuration Propagation from Admin Panel to Storefront
+
+## Problem Statement
+
+Existing Magento monolith is responsible for the whole cycle of data and user workflow.
+Magento Admin Panel contains configuration settings for all areas of the application, including Admin Panel behavior, storefront UI, etc.
+
+With Storefront being a separate service (or set of services), should configuration settings in Magento Admin Panel be propagated to storefront service?
+If yes, in what way?
+
+## Configuration Propagation Guidelines
+
+This document covers general agreements for decisions about configuration propagation from Admin Panel to storefront. 
+A separate design decision should be made and documented for specific use cases.
+
+There are three use cases for the configuration, based on its impact.
+
+### 1. Configuration that impacts Admin Panel behavior
+
+Examples: admin ACLs that allow or deny parts of functionality for the admin user, reindex on update or by schedule.
+
+This type of configuration has nothing to do with Storefront and should not be propagated to storefront, as well as should not be exposed via export API. 
+
+### 2. Configuration that impacts data provided by Storefront
+
+Example: Base Media URL is used for calculation of media image URLs returned by Storefront API.
+
+This type of configuration should be propagated to Storefront in one of the following ways:
+
+1. Pre-calculate final data on back office side and provide final result to the Storefront during synchronization.
+   1. Pros: simpler implementation of Storefront due to eliminated necessity for Storefront to keep knowledge about additional configuration, including data calculation algorithms.
+   2. Cons: massive (up to full) reindexation necessary in case the configuration is changed.
+   
+To choose the right approach in a specific case, consider the following:
+
+1. How frequently the configuration is expected to change?
+   1. Frequent configuration changes leading every time to massive reindexation may be unacceptable.
+   2. Configuration changes expected a few times in the store lifespan may not be worth additional complexity on the Storefront side and full reindexation may be better in this case.
```

Comment:
Lets' consider example with base media URL. Looks like this is not often operation, but it assumes that image will not be valid on Storefront until full indexation. To solve it we need to create redirects, but who will be responsible for that? 

I propose to add one more criterion:
If using configuration on Storefront service allows providing different results with the simple operation (using 1 of 2 fields or concatenation) then it's better to calculate this value on SF side

Reply:
All simple things turn into complex ones with time. And, we all know, that with Magento scopes simple becomes complex due to fallbacks. But I agree the criterion about working store with both old and new data should be made more clear.

Reply:
Media URLs will become even more complex in case an external DAM comes into play. If we decide to support both Magento and external DAM (which have different URLs), we would need to calculate which product and which image is served from where. It's not necessarily how we will implement it (we may decide to not support mixed mode), but most likely such mode will be necessary anyways (at least for transition period). It's mach simpler to offload this work from SF to backoffice and agree that Catalog SF doesn't need to know anything about DAM systems and it just ingests and then serves image URLs.